packages/plugin-ext/src/hosted/node/hosted-plugin-deployer-handler.ts

```diff
@@ -47,14 +47,22 @@ export class HostedPluginDeployerHandler implements PluginDeployerHandler {
         // await first deploy
         await this.frontendPluginsMetadataDeferred.promise;
         // fetch the last deployed state
-        return this.currentFrontendPluginsMetadata;
+        return [...this.currentFrontendPluginsMetadata.values()];
     }
 
     async getDeployedBackendMetadata(): Promise<PluginMetadata[]> {
         // await first deploy
         await this.backendPluginsMetadataDeferred.promise;
         // fetch the last deployed state
-        return this.currentBackendPluginsMetadata;
+        return [...this.currentBackendPluginsMetadata.values()];
+    }
+
+    getDeployedPluginMetadata(pluginId: string): PluginMetadata | undefined {
+        const metadata = this.currentBackendPluginsMetadata.get(pluginId);
```

Comment:
I have a couple of questions.

we cache the model with its id (publisher + name)

- what is happening for development mode where we might change package.json without changing publisher/name
- what is happening if we bump to a new plug-in version

Reply:
It would the same as before. This PR does not change it, caching was there before to avoid deploying duplicate plugins.

> what is happening for development mode where we might change package.json without changing publisher/name

Seems to be a bug, should be the same on master.

> what is happening if we bump to a new plug-in version

So far the plugin system did not apply semver to decide which plugin to deploy and pick the first one. It should be an enhancement request.